src/App.tsx

```diff
@@ -15,49 +20,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+interface ReorderOptions {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const goodsToShow = [...goods];
+
+  goodsToShow.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    goodsToShow.reverse();
+  }
+
+  return goodsToShow;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sortByLength = () => setSortType(SortType.LENGTH);
+
+  const sortByAlphabet = () => setSortType(SortType.ALPHABET);
+
+  const reverse = () => setIsReversed((state) => !state);
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const reorderedGoods = getReorderedGoods(
+    goodsFromServer, { sortType, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
+        <Button
+          onClick={sortByAlphabet}
+          className={
+            classNames('is-info', {
+              'is-light': sortType !== SortType.ALPHABET,
+            })
+          }
         >
           Sort alphabetically
-        </button>
+        </Button>
 
-        <button
-          type="button"
-          className="button is-success is-light"
+        <Button
+          onClick={sortByLength}
+          className={
+            classNames('is-success', {
+              'is-light': sortType !== SortType.LENGTH,
+            })
+          }
         >
           Sort by length
-        </button>
+        </Button>
 
-        <button
-          type="button"
-          className="button is-warning is-light"
+        <Button
+          onClick={reverse}
+          className={
+            classNames('is-warning', {
+              'is-light': !isReversed,
+            })
+          }
         >
           Reverse
-        </button>
+        </Button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || isReversed) && (
```

Comment:
if you will need to change the condition in the future - you will have to look for this peace of code in order to change it. So it is better to create a separate variable for that